### ISSUE TITLE
Improving the way test reports are searched and found

### ISSUE THREAD

**Issue by MarcoMinghini** [INSPIRE-MIF/helpdesk-validator#7]
Currently it is difficult to find "your" test report in the list of test reports, because test reports are sorted by date and time and their names are all very similar (almost all users use the pre-defined title). 

Thus, two possible improvements are the following:

1) **before executing the test**: changing the default naming scheme for the test reports to be generated.
For example, this name can include only the name of the test object (e.g. filename or abbreviated URL), the test object type (metadata records, WFS, etc.) and the date.
In addition, we could allow the user to specify a "tag" or label when launching a test run (e.g. "jrc" or "germany"). This tag could be appended to the test report name, e.g. as "[jrc] <name of test report>".

Regarding the addition of a label, the UI can be improved like in this mockup:

![imagen](https://user-images.githubusercontent.com/10537778/47159704-bf664480-d2ee-11e8-8feb-90ca638c0643.png)

2) **after executing the test**: introducing different ways to filter/sort the test reports, e.g. by date, time, name, conformance class, category (where categories are the headers of the groups of tests such as "Download Services" or "Metadata"), test object type, filename, URL, etc.

An example of filter by date, label, test suite and name is represented in these mockups:

![imagen](https://user-images.githubusercontent.com/10537778/47159821-ec1a5c00-d2ee-11e8-9372-bb462c9771fa.png)

![imagen](https://user-images.githubusercontent.com/10537778/47159832-f50b2d80-d2ee-11e8-82ee-3567ee5fdd47.png)

What are your views, comments and suggestions about this potential improvement?

**Comment by thijsbrentjens** [comment]
Really good idea. The labelling idea is very nice.

**Comment by MarcoMinghini** [comment]
[2017.4 meeting 2018-11-15] the improvement proposal is considered to be useful and nice to have. The idea of adding labels before running the test is very appreciated.

**Comment by frcaldeira** [comment]
This functionality will be very helpful, great job!

**Comment by AntoRot** [comment]
I agree too that improvement would be very useful.

**Comment by IGNCNIG** [comment]
It’s a really good idea.
So, for example, when testing a metadata with the file identifier “spaignLLM”, the name could be: “spaignLLM_metadata_20190117_13:43_[tag]” [UUID_metadata_Date_time]
And for service  (example of WMS http://www.ign.es/wms-inspire/unidades-administrativas?request=GetCapabilities&service=WMS) , the name could be: “Unidades_administrativas_de_España_WMS_20190117_13:43_[tag]” [Title of service_TypeService_Date_time]

**Comment by MarcoMinghini** [comment]
As a follow-up of this work, you can now test a **first draft of the development** in the [staging instance of the Validator](http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp).

In particular:

- before starting a test, you can see a text box named _Tags_ where you can enter your tags. At the moment, this is simply a string (i.e. if you enter something like "jrc, italy" you are not entering two tags, but just one string)
- after running the test, from the _Test reports_ page, you can use the _Filter Test Reports_ bar to search for your tags. Again, this is a full-text search (not a search on each tag).

This is only a draft, and current/future work will also add the chance to filter test reports by date and test suite(s). 

However, **we would like to get your feedback** on how the search of test reports by tag can be improved, since at the moment we believe that it is not fully clear to users that they can search by tags.
Any comment or suggestion is appreciated.

**Comment by alitka** [comment]
I tested and the search via "tag" is working properly. To improve the usability of tag-usage, you could maybe add a "tool tip" (as already implemented for the label), e.g. "The entered tag can be use to filter/sort the test reports."

**Comment by MarcoMinghini** [comment]
Do you have any additional suggestions on how the search of test reports by tag can be improved? We would like to move forward with this - any feedback would be appreciated.

**Comment by MarcoMinghini** [comment]
Since no further feedback was provided, we will simply add a tool tip for 'Tags' (similarly to the tool tip already available for 'Label').
